### PR TITLE
Completed double factorial issue assigned to me

### PR DESCRIPTION
**Pull request by bagladivyang03**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)

### Description

(Write your answer here.)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

https://github.com/TesseractCoding/NeoAlgo/issues/930#issue-694684093

### CODE REVIEW

**Comment by bagladivyang03** [comment]
@Kajol-Kumari @Abhijit2505 @SKAUL05 @siddharth25pandey @HarshCasper 
Pls review and merge it asap :)

**Review by Kajol-Kumari — CHANGES_REQUESTED** [review]
@bagladivyang03 please do the following:
1. Link the related issue in the PR description. 
2. Please provide 1-2 line comment on the top of the file regarding what this code is going to do.

**Comment by bagladivyang03** [comment]
@Kajol-Kumari done
Pls review again asap :)

File: C/cp/doublefactorial.c [hunk 1]
```diff
@@ -0,0 +1,28 @@
+#Double factorial of a non-negative integer n is defined as the product of all the integers from 1 to n that have the same parity (odd or even) as n.
```
Comment (Kajol-Kumari) [inline C/cp/doublefactorial.c]: @bagladivyang03 This is not the correct way for commenting in C. Please use multi-line comment here `/* */`

**Review by Kajol-Kumari — CHANGES_REQUESTED** [review]

**Comment by bagladivyang03** [comment]
@Kajol-Kumari yeah sorry because of using the python i had a habit of that
Now completed in correct way xD

**Review by Kajol-Kumari — APPROVED** [review]
LGTM :+1:

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@bagladivyang03 What you have added is Iterative Solution please add Recursive Solution too.

**Comment by bagladivyang03** [comment]
@iamrajiv added recursive soln too just check once pls :)

File: C/cp/doublefactorial.c [hunk 2]
```diff
@@ -0,0 +1,62 @@
+#include <stdio.h>
+int doublefactorial(int n){
```
Comment (iamrajiv) [inline C/cp/doublefactorial.c]: @bagladivyang03 rename this to `iterative _doublefactorial`

File: C/cp/doublefactorial.c [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    for(int i=n;i>0;i-=2){
+    if(n==0 || n==1)
+        return res;
+    else
+         res = res*i;
+    }
+
+}
+
+int recdoublefactorial(int n){
```
Comment (iamrajiv) [inline C/cp/doublefactorial.c]: @bagladivyang03 rename this to `recursive _doublefactorial`

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@bagladivyang03 Format your code using code formatter.

**Comment by bagladivyang03** [comment]
@iamrajiv pls check now

**Comment by bagladivyang03** [comment]
@atarax665 pls do check and merge asap

**Review by iamrajiv — APPROVED** [review]

**Comment by bagladivyang03** [comment]
@atarax665 pls could u review

File: C/README.md [hunk 4]
```diff
@@ -21,6 +21,7 @@
 - [Randomized selection algorithm](search/randomized_selection_algorithm.c)
 - [Sum of Triplet](cp/sum_of_triplet.c)
 
+
```
Comment (atarax665) [inline C/README.md:25]: Update Readme.md

**Review by atarax665 — CHANGES_REQUESTED** [review]

File: C/README.md (hunk 4 above)
Comment (bagladivyang03) [reply C/README.md:25]: its showing conflicts could u pls hlp me

**Comment by bagladivyang03** [comment]
@atarax665 pls merge now asap :)

**Comment by bagladivyang03** [comment]
i have  updated the readme too

**Pull request merged**